C#/C#.sublime-syntax

```diff
@@ -1665,10 +1665,11 @@ contexts:
       scope: invalid.illegal.unexpected-character-in-placeholder.cs
 
   string_placeholder_format:
-    - match: '\s*(?:(,)\s*(-?\d+)\s*)?'
+    - match: '\s*(?:(,)\s*((-?)\d+)\s*)?'
       captures:
         1: punctuation.separator.arguments.cs
-        2: constant.numeric.integer.decimal.formatting.cs
```

Comment:
out of curiosity, was the removal of the `.formatting` specialization deliberate? As far as I can tell, it is now no longer possible for a color scheme to target the formatting number/padding amount separately from the argument number.
i.e. in the example `string multiple = String.Format("0x{0:X} |{0,-15}| -{0,5:N2}-", 12);`, the `0` after `{` has the same scope as the `15` and the `5`. Should we start another RFC for how to scope these?

For reference, the above example outputs:
`0xC |12             | - 12.0-`

So we can see that the `5`/`-15` reflects left/right padding. So I propose `meta.formatting.padding-amount` for this, which can be combined with the `meta.number` and/or `constant.numeric` for the color scheme to target.

Reply:
Removed the formatting as I strictly followed the modified `meta.number constant.numeric` guidelines which basically remove all kinds of sematic relation from `constant.numeric` due to limiting the 3rd-level scope to `base|value|suffix` in the first place.

We may move the `formatting` to the `meta.number. ... ` scope if requried, as the meta scope now holds sematic information. Not sure if we want to create `constant.numeric.value.formatting` as I already saw things like `constant.numeric.value.exponent` to denote exponent part of floats.

I'd appreciate a common sense how to handle such formatting literals as we can find them in several languages. I agree with @wbond about having numbers scoped as such in formatting placeholders may be helpful and you are right, they should propably be distinguishable from "normal" numbers in some way.

Tokens like `{0:X}` are assigned  `constant.other.placeholder`. The `0` within therefore can be identified by `constant.other.placeholder constant.numeric`. I could also imagine to use `meta.placeholder constant.numeric` for such kinds of tokens, which whould match the scheme of `meta.interpolation` and avoid stacking different `constant.` scopes.

Not sure whether it was suitable for C-style formats though, which just use `%02d` or `%02.04f`. They just receive `constant.other.placeholder` atm and I guess it is the reason for python to scope `{0}` and friends the same way.

Reply:
Thanks, I like your reasoning :) I will create an RFC issue about it when I can make time, unless someone beats me to it :)